discovery_engine_core/web-api/src/elastic.rs

```diff
@@ -165,3 +166,23 @@ struct Hit<T> {
 struct Total {
     value: usize,
 }
+
+pub(crate) mod serde_embedding_as_vec {
+    use ndarray::Array;
+    use serde::{Deserialize, Deserializer, Serialize, Serializer};
+    use xayn_discovery_engine_ai::Embedding;
+
+    pub(crate) fn serialize<S>(embedding: &Embedding, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        embedding.iter().collect::<Vec<_>>().serialize(serializer)
```

Comment:
i like that we can keep the `Embedding` wrapper in `ElasticDocumentData`:+1: the issue of additional allocations is still the same though, we could use the same solution proposed in https://github.com/xaynetwork/xayn_discovery_engine/pull/644#discussion_r994303689:
```suggestion
        embedding.into_vec().serialize(serializer)
```

Reply:
I used serializing sequence approach described here https://serde.rs/impl-serialize.html#serializing-a-sequence-or-map, please check again :)